src/test/mir-opt/matches_reduce_branches.foo.PreCodegen.before.64bit.mir

```diff
@@ -3,8 +3,10 @@
 fn foo(_1: Option<()>) -> () {
     debug bar => _1;                     // in scope 0 at $DIR/matches_reduce_branches.rs:7:8: 7:11
     let mut _0: ();                      // return place in scope 0 at $DIR/matches_reduce_branches.rs:7:25: 7:25
+    let mut _2: isize;                   // in scope 0 at $DIR/matches_reduce_branches.rs:8:22: 8:26
 
     bb0: {
-        return;                          // scope 0 at $DIR/matches_reduce_branches.rs:11:2: 11:2
+        _2 = discriminant(_1);           // scope 0 at $DIR/matches_reduce_branches.rs:8:22: 8:26
```

Comment:
It's weird that this comes back, we should be removing the ZST assignment even earlier than before.